lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/EvaluatorRuntime.cs

```diff
@@ -254,21 +253,12 @@ private void OnException(Exception e)
 
                 try
                 {
-                    using (var memStream = new MemoryStream())
-                    {
-                        var formatter = new BinaryFormatter();
-                        formatter.Serialize(memStream, e);
-                        errorBytes = memStream.ToArray();
-                    }
+                    errorBytes = ByteUtilities.SerializeToBinaryFormat(e);
                 }
                 catch (SerializationException se)
```

Comment:
For Tasks, we should use `NonSerializableTaskException` instead (which I forgot to do in this iteration, I'll update). Given the need to differentiate the different kinds of Exceptions to throw, I think it's best to not generalize and do more than the `static` function advertises, which only promises to serialize/deserialize and object without specifying "where" it's serializing/deserializing.